src/model/consent/consent.ts

```diff
@@ -0,0 +1,148 @@
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop
+ files are distributed onan 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+/*
+ * Unlike MySQL, SQLite (testing DB) doesn't support Timestamp type natively.
+ * It returns ISO strings using the inbuilt Date() function.
+ * Thus, typecasting of MySQL Timestamp to Date object for insertion/retrieval
+ * for 'createdAt' field and testing for the same may be required in future.
+ */
+
+import { NotFoundError } from '../errors'
+import Knex from 'knex'
+
+/*
+ * Interface for Consent resource type
+ */
+export interface Consent {
+  id: string;
+  initiatorId: string;
+  participantId: string;
+  credentialId?: string;
+  credentialType?: string;
+  credentialStatus?: string;
+  credentialPayload?: string;
+  credentialChallenge?: string;
+  createdAt?: Date;
+}
+
+/*
+ * Class to abstract Consent DB operations
+ */
+export class ConsentDB {
+  // Knex instance
+  private Db: Knex
+  // Nullable fields which don't allow for explicit null upsert
+  private static nullProtectedProps: Record<string, boolean> = {
+    credentialId: true,
+    credentialType: true,
+    credentialStatus: true,
+    credentialPayload: true,
+    credentialChallenge: true
+  }
+
+  public constructor (dbInstance: Knex) {
+    this.Db = dbInstance
+  }
+
+  // Add initial Consent parameters
+  public async register (consent: Consent): Promise<number> {
+    // Returns array containing number of inserted rows
+    const insertCount: number[] = await this
+      .Db<Consent>('Consent')
+      .insert(consent)
+
+    return insertCount[0]
+  }
+
+  // Update Consent credential
+  // Only non-null Consent fields are updated
+  public async updateCredentials (consent: Consent): Promise<number> {
+    const validatedConsent = {}
+
+    for (const key in consent) {
```

Comment:
If I have an interface `Thing` with properties `a: string` and `b: string?`, and I create this interface as `Thing({ a: "a" })`, when I iterate, will I see `key` values of both `"a"` and `"b"` ? Or just `"a"`? If just a, is that OK?

Reply:
in TS you can't have an instance of an interface, but only the object which conforms to interface declarations. So the answer is: for object {a: 'a'} which conforms to Thing interface you will have just `a`.